src/legacy/core_plugins/console/public/np_ready/application/models/sense_editor/curl.ts

```diff
@@ -0,0 +1,204 @@
+    } else if (line.substr(0, 1) === '"') {
+      line = line.substr(1);
+      state = 'DOUBLE_QUOTE';
+    } else {
+      state = 'UNQUOTED';
+    }
+  }
+
+  // Body is finished - append to output with final LF
+  function addBodyToOut() {
+    if (body.length > 0) {
+      out.push(body.join(''));
+      body = [];
+    }
+    state = 'LF';
+    out.push('\n');
+  }
+
+  // If the pattern matches, then the state is about to change,
+  // so add the capture to the body and detect the next state
+  // Otherwise add the whole line
+  function consumeMatching(pattern: string | RegExp) {
+    const result = line.match(pattern);
+    if (result) {
+      body.push(result[1]);
+      line = line.substr(result[0].length);
+      detectQuote();
+    } else {
+      body.push(line);
+      line = '';
+    }
+  }
+
+  function parseCurlLine() {
+    let verb = 'GET';
+    let request = '';
+    let result;
+    if ((result = line.match(CurlVerb))) {
+      verb = result[1];
+    }
+
+    // JS regexen don't support possessive quantifiers, so
+    // we need two distinct patterns
+    const pattern = HasProtocol.test(line) ? CurlRequestWithProto : CurlRequestWithoutProto;
+
+    if ((result = line.match(pattern))) {
+      request = result[1];
+    }
+
+    out.push(verb + ' /' + request + '\n');
+
+    if ((result = line.match(CurlData))) {
+      line = line.substr(result[0].length);
+      detectQuote();
+      if (EmptyLine.test(line)) {
+        line = '';
+      }
+    } else {
+      state = 'NONE';
+      line = '';
+      out.push('');
+    }
+  }
+
+  while (nextLine()) {
+    if (state === 'SINGLE_QUOTE') {
+      consumeMatching(ClosingSingleQuote);
+    } else if (state === 'DOUBLE_QUOTE') {
+      consumeMatching(ClosingDoubleQuote);
+      unescapeLastBodyEl();
+    } else if (state === 'UNQUOTED') {
+      consumeMatching(EscapedQuotes);
+      if (body.length) {
+        unescapeLastBodyEl();
+      }
+      if (state === 'UNQUOTED') {
+        addBodyToOut();
+        line = '';
+      }
+    }
+
+    // the BODY state (used to match the body of a Sense request)
+    // can be terminated early if it encounters
+    // a comment or an empty line
+    else if (state === 'BODY') {
+      if (Comment.test(line) || EmptyLine.test(line)) {
+        addBodyToOut();
+      } else {
+        body.push(line);
+        line = '';
+      }
+    } else if (EmptyLine.test(line)) {
+      if (state !== 'LF') {
+        out.push('\n');
+        state = 'LF';
+      }
+      line = '';
+    } else if ((matches = line.match(Comment))) {
+      out.push('#' + matches[1] + '\n');
+      state = 'NONE';
+      line = '';
+    } else if (LooksLikeCurl.test(line)) {
+      parseCurlLine();
+    } else if ((matches = line.match(SenseLine))) {
+      out.push(matches[1] + ' /' + matches[2] + '\n');
+      line = '';
+      state = 'BODY';
+    }
+
+    // Nothing else matches, so output with a prefix of !!! for debugging purposes
```

Comment:
Looks like this comment is stale?
```suggestion
    // Nothing else matches, so output with a prefix of ### for debugging purposes
```